src/App.js

```diff
@@ -1,25 +1,87 @@
-import logo from './logo.svg';
-import './App.css';
+import { useState, useEffect } from 'react';
+import { nanoid } from 'nanoid';
+// import { localItems } from './db';
+import './styles.css';
+import ListItem from './components/ListItem';
+import Header from './components/Header';
+import AddItem from './components/AddItem';
+import Subheader from './components/Subheader';
+import Fetch from './components/Fetch';
+import ToggleButton from './components/Toggle';
 
 function App() {
+  // const [articles, setArticles] = useState(localItems);
+  const [articles, setArticles] = useState(loadFromLocal('products') ?? []);
+  const [searchValue, setSearchValue] = useState('');
+
+  const [language, setLanguage] = ToggleButton();
+
+  useEffect(() => {
+    saveToLocal('products', articles);
+  }, [articles]);
+
+  function loadFromLocal(key) {
+    try {
+      return JSON.parse(localStorage.getItem(key));
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  function saveToLocal(key, data) {
+    localStorage.setItem(key, JSON.stringify(data));
+  }
+
+  function handleOnDelete(articleId) {
+    setArticles(articles.filter(article => article._id !== articleId));
+  }
+  function handleOnAdd(articleName) {
+    const newArticleName = {
+      _id: nanoid(),
+      _type: 'shopping.item',
+      category: { _type: 'ref', _ref: 'c2hvcHBpbmcuY2F0ZWdvcnk6MA==' },
+      name: { en: articleName, de: '' },
+    };
+    setArticles([...articles, newArticleName]);
+  }
+
+  function handleOnSearch(shoppingItem) {
+    if (articles.find(article => article._id === shoppingItem._id)) {
+      alert('The article is already on your list');
+      setSearchValue('');
+    } else {
+      setSearchValue('');
+      setArticles([...articles, shoppingItem]);
+    }
+  }
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div className="App__container">
+      <Header text="Shopping List" />
+      <ListItem curry={articles} onDelete={handleOnDelete} />
```

Comment:
Please remove the `curry` from all within your code and rename them for what they really are. For example here it would be `articles`
```suggestion
      <ListItem articles={articles} onDelete={handleOnDelete} />
```